R/assess_r_cmd_check.R

```diff
@@ -0,0 +1,25 @@
+#' Assess a package's results from running R CMD check 
+#'
+#' @export
+assess_r_cmd_check <- function(x, ...) {
+  UseMethod("assess_r_cmd_check")
+}
+
+attributes(assess_r_cmd_check)$column_name <- "r_cmd_check"
+attributes(assess_r_cmd_check)$label <- "Package check results"
+
+assess_r_cmd_check.default <- function(x, ...) {
+  pkg_metric_na(message = "Source code not available to run R CMD check on.")
+}
+assess_r_cmd_check.pkg_source <- function(x, ...) {
+  pkg_metric(sapply(x$r_cmd_check[c("notes","errors","warnings")], length), 
+             class = "pkg_metric_r_cmd_check")
+}
+
+assess_r_cmd_check.pkg_cran_remote <- function(x, ...) {
+  pkg_metric_todo("Assessment of R CMD check on remote pkg refs is not yet implemented but will be in the future")
+}
+
+assess_r_cmd_check.pkg_bioc_remote <- function(x, ...) {
+  pkg_metric_todo("Assessment of R CMD check on remote pkg refs is not yet implemented but will be in the future")
+}
```

Comment:
In order for `assess_r_cmd_chec(<pkg_ref>)` to work as a stand-alone function, we need to export all of these methods. As it is now, the assessment will only work if it's called through a higher level function like `pkg_assess`.

```suggestion
#' @export
assess_r_cmd_check.default <- function(x, ...) {
  pkg_metric_na(message = "Source code not available to run R CMD check on.")
}

#' @export
assess_r_cmd_check.pkg_source <- function(x, ...) {
  pkg_metric(sapply(x$r_cmd_check[c("notes","errors","warnings")], length), 
             class = "pkg_metric_r_cmd_check")
}

#' @export
assess_r_cmd_check.pkg_cran_remote <- function(x, ...) {
  pkg_metric_todo("Assessment of R CMD check on remote pkg refs is not yet implemented but will be in the future")
}

#' @export
assess_r_cmd_check.pkg_bioc_remote <- function(x, ...) {
  pkg_metric_todo("Assessment of R CMD check on remote pkg refs is not yet implemented but will be in the future")
}
```